Engine/Fuego/MacOS/Metal/MetalContext.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Renderer/GraphicsContext.h"
+#include <Metal/Metal.hpp>
+#include <QuartzCore/CAMetalLayer.hpp>
+#include <QuartzCore/CAMetalLayer.h>
+#include <QuartzCore/QuartzCore.hpp>
+#include <simd/simd.h>
+
+
+namespace Fuego
+{
+class MetalContext : public GraphicsContext
+{
+   public:
+    MetalContext(void* nsWindow);
+    virtual ~MetalContext() override;
+
+    virtual bool Init() override;
+    virtual void SwapBuffers() override;
```

Comment:
Split to acquire and present. Present can be done only in graphics queue. Acquire is a function of swapchain. Only one image at a time or not more than swapchain image count.